### PR TITLE
loadbalancer[1]: add initial implementation of LoadBalancer interface

### PR DESCRIPTION
**Pull request by nicolehanjing**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This is an initial PR of implementing the load balancer. 
implemented several functions:
- GetLoadBalancer
- GetLoadBalancerName
- UpdateLoadBalancer
- EnsureLoadBalancerDeleted

some TODOs for follow-up PRs:
- implement `EnsureLoadBalancer` function
- handle cluster tags
- external LB/ NLB
- `GetLoadBalancerName` may need improvements

**Which issue(s) this PR fixes**:

Part of #125

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add initial implementation of loadbalancer for the v2 provider.
```

### CODE REVIEW

File: pkg/providers/v2/loadbalancer.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	v1 "k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/klog/v2"
+)
+
+// newLoadBalancer returns an implementation of cloudprovider.LoadBalancer
+func newLoadBalancer(region string, creds *credentials.Credentials) (cloudprovider.LoadBalancer, error) {
```
Comment (HaibaraAi96) [inline pkg/providers/v2/loadbalancer.go:36]: @andrewsykim I'm just following the pattern for implementing the LoadBalancer interface
For the functions to implement, I leave EnsureLoadBalancer() and DeleteLoadBalancer() empty and still try to gain a better understanding of the implementation
For other functions, I still have a lot of TODOs and probably need more contexts, so I might change my code a lot very soon, this just shows the basic structure of my implementation

**Comment by nckturner** [comment]
A couple points for load balancer in V2.  We're planning on moving classic load balancers to a deprecated state so that new services use NLB's by default, and we currently have NLB code split between a target IP mode, which is going into the "aws-load-balancer-controller" and the existing NLB instance mode, which is in the legacy cloud provider.  We'd like to move both NLB target types to the aws-load-balancer-controller, so all new services would be delegated to that, unless they specifically requested legacy behavior.  In terms of what the legacy classic load balancer support looks like, I'm guessing what we're planning here will be fine, which from my perspective is basically the existing behavior with support for friendly names? 

/cc @M00nF1sh

**Comment by nicolehanjing** [comment]
> A couple points for load balancers in V2. We're planning on eventually moving classic load balancers to a deprecated state so that new services use NLB's by default, and we currently have NLB code split between a target IP mode, which is going into the "aws-load-balancer-controller" and the existing NLB instance mode, which is in the legacy cloud provider. We'd like to move both NLB target types to the aws-load-balancer-controller, so all new services would be delegated to that, unless they specifically requested legacy behavior. In terms of what the legacy classic load balancer support looks like, I'm guessing what we're planning here will be fine, which from my perspective is basically the existing behavior with support for friendly names?
> 
> /cc @M00nF1sh

yeah most of the behavior refers to legacy implementation but we need to support a friendly name for the load balancer
I think we would want to have clusterName and serviceName included in the LB name, but also make this name unique with a maximum of 32 characters

**Pull request closed without merging**